src/JuMP.jl

```diff
@@ -579,17 +582,70 @@ function getNumSOCRows(m::Model)
     return numSOCRows
 end
 
-function getdual(c::ConstraintRef{Model,SOCConstraint})
-    numBndRows = getNumBndRows(c.m)
-    numSOCRows = getNumSOCRows(c.m)
-    if length(c.m.conicconstrDuals) != (MathProgBase.numlinconstr(c.m) + numBndRows + numSOCRows)
-        error("Dual solution not available. Check that the model was properly solved and no integer variables are present.")
+# Returns the number of semidefinite constraints
+function getNumSDPRows(m::Model)
+    numSDPRows = 0
+    for con in m.sdpconstr
+        n = size(con.terms, 1)
+        numSDPRows += convert(Int, n*(n+1)/2)
     end
-    return c.m.conicconstrDuals[
-        c.m.constrDualMap[MathProgBase.numlinconstr(c.m) + numBndRows + c.idx]]
+    return numSDPRows
 end
 
+# Returns the number of symmetry-enforcing constraints for SDP constraints
+function getNumSymRows(m::Model)
+    sum(map(length, m.sdpconstrSym))
+end
 
+function getconicdualaux(m::Model, idx, issdp)
+    numLinRows = MathProgBase.numlinconstr(m)
+    numBndRows = getNumBndRows(m)
+    numSOCRows = getNumSOCRows(m)
+    numSDPRows = getNumSDPRows(m)
+    numSymRows = getNumSymRows(m)
+    if length(m.conicconstrDuals) != (numLinRows + numBndRows + numSOCRows + numSDPRows + numSymRows)
+        error("Dual solution not available. Check that the model was properly solved and no integer variables are present.")
+    end
+    offset = numLinRows + numBndRows
+    if issdp
+        offset += length(m.socconstr)
+    end
+    dual = m.conicconstrDuals[m.constrDualMap[offset + idx]]
+    if issdp
+        offset += length(m.sdpconstr)
+        symdual = m.conicconstrDuals[m.constrDualMap[offset + idx]]
+        dual, symdual
+    else
+        dual
+    end
+end
+
+function getdual(c::ConstraintRef{Model,SOCConstraint})
+    getconicdualaux(c.m, c.idx, false)
+end
+
+function getdual(c::ConstraintRef{Model,SDConstraint})
+    dual, symdual = getconicdualaux(c.m, c.idx, true)
+    n = size(c.m.sdpconstr[c.idx].terms, 1)
+    X = Matrix{eltype(dual)}(n, n)
+    @assert length(dual) == convert(Int, n*(n+1)/2)
+    idx = 0
+    for i in 1:n
+        for j in i:n
+            idx += 1
+            X[j,i] = X[i,j] = dual[idx] / sqrt(2)
+        end
+    end
+    @assert length(symdual) == length(c.m.sdpconstrSym[c.idx])
+    idx = 0
+    for (i,j) in c.m.sdpconstrSym[c.idx]
```

Comment:
This is a bit tricky to explain without going into the details and I have not found a reference (classical reference only consider the symmetric case).
Therefore I have added a detailed explanation of why it is the correct way to do so